### PR TITLE
[caffe2][script] Misc. fixes and improvements

### PR DESCRIPTION
**Pull request by zdevito**
* condition if
* True/False literals
* and, or, not
* 0-output expressions, like print
* _ is given a fresh name
* x.foo(...) is desugared to foo(x,...)
* +=, *=

### CODE REVIEW

File: caffe2/contrib/script/compiler.cc [hunk 1]
```diff
@@ -251,6 +276,9 @@ struct DefCompiler {
     } else if (type_ident == "LL") {
       dtype->set_i(TensorProto_DataType_INT64);
       value->set_i(v);
+    } else if (type_ident == "b") {
```
Comment (jamesr66a) [inline caffe2/contrib/script/compiler.cc]: Is this only an implementation detail in the compiler or in the language can I pass, for ex, 1234b and that will emit a ConstantFill(type=bool, value=1i)?

**Review by jamesr66a — APPROVED** [review]
LGTM! Thanks!

File: caffe2/contrib/script/compiler.cc (hunk 1 above)
Comment (zdevito) [reply caffe2/contrib/script/compiler.cc]: It is an implementation detail, True generates (TK_CONST 1 b). I will probably remove this tomorrow in favor of making it (TK_CONST "True" b), (TK_CONST "1.0" "f"), (TK_CONST "34" "I"), etc.

**Comment by pietern** [comment]
@zdevito The android failure is real in this case. Just like stoi and stod it needs to use the caffe2 wrapper for those as they don't exist in Android's C++11.

**Pull request closed without merging**